src/api/routes/ws/socket-io.ts

```diff
@@ -17,8 +17,39 @@ import {
   getTxFromDataStore,
   parseDbTx,
 } from '../../controllers/db-controller';
-import { isProdEnv, logError, logger } from '../../../helpers';
+import { isProdEnv, isValidPrincipal, logError, logger } from '../../../helpers';
 import { WebSocketPrometheus } from './metrics';
+import { isValidTxId } from '../../../api/query-helpers';
+
+function getInvalidSub(subscriptions: Topic | Topic[]): undefined | string {
+  const isSubValid = (sub: Topic): undefined | string => {
+    if (sub.includes(':')) {
+      const txOrAddr = sub.split(':')[0];
+      const value = sub.split(':')[1];
+      switch (txOrAddr) {
+        case 'address-transaction':
+        case 'address-stx-balance':
+          return isValidPrincipal(value) ? undefined : sub;
+        case 'transaction':
+          return isValidTxId(value) ? undefined : sub;
+        default:
+          return sub;
+      }
+    }
+    switch (sub) {
+      case 'block':
+      case 'mempool':
+      case 'microblock':
+        return undefined;
+      default:
+        return sub;
+    }
+  };
+  if (!Array.isArray(subscriptions)) return isSubValid(subscriptions);
+  const validatedSubs = subscriptions.map(isSubValid);
+  const invalidSub = validatedSubs.find(validSub => typeof validSub === 'string');
```

Comment:
You should use `filter` instead of `find` in order to return all invalid topics. This way you can list them all in the returned error as a comma-separated string.